### DISCUSSION TITLE
Help: No parser could be inferred for file

### DISCUSSION THREAD

**Discussion by Permanently** [prettier/plugin-pug#463] [Q&A] [answered]
Hello! About 10 minutes ago I blundered (#462) by not seeing the Discussions tab. So, here goes again.

I'm trying to use the Pug plugin for basic functionality, and have run [the top two commands on the docs](https://prettier.github.io/plugin-pug/guide/), but to no avail. Instead, when I run `npx prettier --write "**/*.pug"`, I get this:

```
[error] No parser could be inferred for file "/path/to/file/here/page.pug".
``` 

I'm not sure if I'm supposed to have a configuration or anything? I've got `prettier` and `@prettier/plugin-pug` in my devDependencies, and `npx prettier --write "**/*.js"` works fine, so I'm not sure if I've done something wrong. Or, maybe it's my tiredness again. I'm not sure at this rate.

If anyone has any suggestions, I'd appreciate them!

**Comment by Shinigami92** [comment]
I think prettier v3 changed something so you need to load the plugin slightly differently 
Check their docs

**Comment by Permanently** [comment, marked as answer]
Thanks- [seems v3 has changed things](https://prettier.io/docs/en/plugins#using-plugins) after all. The command is:

`npx prettier --write "**/*.pug" --plugin="@prettier/plugin-pug"`
